### ISSUE TITLE
extracted_features might different from the original graph2vec

### ISSUE THREAD

**Issue by ZiruiYan** [benedekrozemberczki/graph2vec#14]
The original graph2vec uses the feature both prev and next extracted features in WL
<img width="758" alt="image" src="https://user-images.githubusercontent.com/55161403/65053747-75d90c00-d93a-11e9-8821-0a71d8ef43e8.png">
But I think this implementation just uses prev extracted features
<img width="611" alt="image" src="https://user-images.githubusercontent.com/55161403/65054252-4676cf00-d93b-11e9-9cd1-fcbecbcabd30.png">

**Comment by ZiruiYan** [comment]
In training, graph2vec just use center not the nei_list, so the set of words will be same. However, I wonder whether the order of extracted_features, which is different in two versions, will hurt the results.

**Comment by shaulz** [comment]
Hi ZiruiYan,
Not sure if this is related, but when I execute on a provided sample, results are not good.
To validate the result, I have added the following code:
`def test(model, document_collections):

    ranks = []

    second_ranks = []

    for doc_id in range(len(document_collections)):

        inferred_vector = model.infer_vector(document_collections[doc_id].words)

        sims = model.docvecs.most_similar([inferred_vector], topn=len(model.docvecs))

        rank = [docid for docid, sim in sims].index("g_" + str(doc_id))

        ranks.append(rank)

        second_ranks.append(sims[1])

    print(sorted(collections.Counter(ranks).items())`

Basically inferred vectors should be most similar to themselves, (0,50)   would be ideal. What I am getting is absolutely random:
[(0, 1), (1, 2), (2, 3), (4, 1), (5, 2), (8, 2), (10, 2), (11, 2), (14, 1), (16, 3), (19, 1), (21, 1), (22, 2), (23, 1), (24, 1), (25, 2), (27, 1), (28, 2), (29, 1), (32, 2), (34, 1), (35, 3), (36, 1), (37, 1), (39, 1), (40, 3), (41, 1), (43, 1), (44, 1), (45, 2), (50, 3)]

Can you provide a fix for the code, so it will work same as original?

**Comment by benedekrozemberczki** [comment]
The provided samples are synthetic data.For inferring you have to use a large learning rate.

**Comment by shaulz** [comment]
Can you please point to the data set and learning rate to be used so the inferring will show reasonable results?

**Comment by benedekrozemberczki** [comment]
Synthetic data means ER graphs. A learning rate above 0.05 helps.

**Comment by shaulz** [comment]
So if I try on nci1 set from the original paper and learning rate 0.05, inferring should be OK right?
Do you have any not synthetic data sets in the json format as the software expects?

**Comment by benedekrozemberczki** [comment]
On a server in Edinburgh, I will look it up for you, but doing the
transformation is not so complicated.

**Comment by shaulz** [comment]
Thanks!!! Really appreciate this. Also want to see the inference working. Do not know if authors of the original paper tried to check the inference.

**Comment by benedekrozemberczki** [comment]
I know someone who tried and for them the mean problem was the:

1. Learning rate.
2. Making sure that the feature spaces overlap.

**Comment by shaulz** [comment]
Hi Benedek, good news, found a bug in my test procedure.
Now I am getting almost perfect inferring results on the synthetic set: [(0, 50), (1, 1)] 
using parameters --learning-rate 0.05 --down-sampling 0.001 --epochs 500

The correct test code:
`def test(model, document_collections):

    ranks = []

    second_ranks = []

    for doc_id in range(len(document_collections)):

        inferred_vector = model.infer_vector(document_collections[doc_id].words)

        sims = model.docvecs.most_similar(positive=[inferred_vector], topn=len(model.docvecs))

        rank = [docid for docid, sim in sims].index(document_collections[doc_id].tags[0])

        ranks.append(rank)

        second_ranks.append(sims[1])

    print(sorted(collections.Counter(ranks).items()))`

**Comment by shaulz** [comment]
Something unrelated, suppose my node data is multidimensional i.e. has more than one label. 
Any idea of how to use graph2vec in such case? Of course I can run separately on each label and merge results into a single TaggedDocument before calling Doc2Vec. Any other options?
